app/client/components/organizationWrapper.jsx

```diff
@@ -0,0 +1,46 @@
+import React, {Component} from 'react';
+import { connect } from 'react-redux';
+import {compose} from 'redux';
+import { updateActiveOrgRequest } from '../reducers/auth/actions'
+import { parseQuery, generateQueryString } from './../utils'
+
+import { browserHistory } from '../app'
+
+const OrganizationWrapper = Component => class extends Component {
+
+    componentDidMount() {
+        //load organisation if param is not matching currently active org
+        let query_params = parseQuery(location.search)
+        if(query_params["org"] && this.props.login.organisationId != query_params["org"]){
+            this.props.updateActiveOrgRequest(query_params["org_name"], query_params["org"])
+        } else {
+            browserHistory.push({
+                search: generateQueryString({org_name: this.props.login.organisationName})
+            })
```

Comment:
This isn't working for me on the TransferAccountList Page...

Something like this works `browserHistory.push(location.pathname + generateQueryString({org_name: this.props.login.organisationName}))`

<img width="1125" alt="Screen Shot 2020-01-21 at 7 01 22 pm" src="https://user-images.githubusercontent.com/27039316/72821236-48ea8c00-3c81-11ea-867d-aa00a6bc9e1d.png">

Reply:
The HOC is clean though!

Reply:
hmmm. did you apply the wrapper to the /accounts component? I just added the wrapper to more of the routes including the transfer account list and it should work now for that page.

Reply:
ok nice... just found another small issue. The HOC is reloading the URL page, so when you click back, it goes from `/accounts/126?org_name=Grassroots%20Economics&org=2` to `/accounts/126`. Not sure how to fix this off the top of my head, but might need to rethink how the HOC is structured...

Reply:
good catch! just fixed it. had to use history.replace instead of history.push because push was adding another path into the history that users were able to traverse through when pressing back

Reply:
nice